internal/kafka/internal/services/kafka.go

```diff
@@ -1376,3 +1421,21 @@ func (k *kafkaService) IsQuotaEntitlementActive(kafkaRequest *dbapi.KafkaRequest
 
 	return quotaService.IsQuotaEntitlementActive(kafkaRequest)
 }
+
+type ClusterSizeCount struct {
+	SizeId string
+	Count  int32
+}
+
+func (k *kafkaService) GetKafkaSizeCountOfEnterpriseCluster(clusterID string) ([]*ClusterSizeCount, error) {
+	dbConn := k.connectionFactory.New()
+	var sizeCountPerCluster []*ClusterSizeCount
+	if err := dbConn.Model(&dbapi.KafkaRequest{}).
+		Select("size_id, count(1) as Count").
+		Group("size_id").
+		Where("cluster_id = ?", clusterID).
```

Comment:
don't we need to also check if the cluster_id is an enterprise cluster?

Reply:
Asking this because the method name `GetKafkaSizeCountOfEnterpriseCluster` has enterprise on it, and there is no where we are checking if it is an enterprise cluster